### PR TITLE
Added a third shared library example and organize the code.

### PR DESCRIPTION
**Pull request by alex4o**
The purpose of the third example is to show, which functions are available to call from each function, and how that changes. As CHERI is using the PCC as a reference of the place where the PLT resides. Thus scanning the PCC reveals all of the accessible functions (and also static variables). 

Side note: I realized that CHERI gives us something akin to reflection for C, which is interesting and might be worth looking into.

### CODE REVIEW

**Comment by ltratt** [comment]
> The purpose of the third example is to show, which functions are available to call from each function, and how that changes. As CHERI is using the PCC as a reference of the place where the PLT resides. Thus scanning the PCC reveals all of the accessible functions (and also static variables).

I'm not sure I understand this. Is the example program expected to succeed or fail or ... ? A comment at the top of the file explaining what it's doing is almost certainly going to be a good idea.

**Comment by alex4o** [comment]
It is supposed to succeed, and show you how things look like in practice.

**Comment by ltratt** [comment]
Shouldn't we also add a second *negative* example that shows what happens if you access a function outside of your compartment?

**Comment by alex4o** [comment]
It turn's out I had forgotten to add the new example in. So the new commit adds the example with the relevant description.

File: shared_objects/unexported_function.c [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+#include "include/unexported_function.h"
+#include "include/find_sentries.h"
+
+__attribute__ ((visibility ("hidden")))
+void do_work() {
+	printf("Doing work\n");
+}
+
+int test() {
+
```
Comment (ltratt) [inline shared_objects/unexported_function.c:19]: Many blank lines!

**Comment by ltratt** [comment]
At the moment these are interesting examples for humans, in the sense that you can run them, but I wonder if we can make them *tests*: i.e. check that we can't reference / call / whatever another function? I think that would make their intent clearer.

File: shared_objects/unexported_function.c (hunk 1 above)
Comment (alex4o) [reply shared_objects/unexported_function.c]: I removed them, but I very much like my blank lines.

Comment (ltratt) [reply shared_objects/unexported_function.c]: :p

**Comment by alex4o** [comment]
Done it is an automatic test using asserts now. And the printing of the table makes it clearer what you need to spot.

**Comment by ltratt** [comment]
Please squash.

**Comment by alex4o** [comment]
I didn't squash them, but I reworded them.

**Pull request merged**